Sources/LanguageServerProtocol/SupportTypes/RegistrationOptions.swift

```diff
@@ -37,9 +28,20 @@ public struct TextDocumentRegistrationOptions: RegistrationOptions, Hashable {
     self.documentSelector = documentSelector
   }
 
-  public func encodeIntoLSPAny(dict: inout [String: LSPAny]) {
-    guard let documentSelector = documentSelector else { return }
-    dict["documentSelector"] = documentSelector.encodeToLSPAny()
+  public init?(fromLSPDictionary dictionary: [String : LSPAny]) {
+    var documentSelector: DocumentSelector?
+
+    if let value = dictionary["documentSelector"] {
+      documentSelector = DocumentSelector(fromLSPArray: value)
+    }
+
+    self.documentSelector = documentSelector
```

Comment:
I think this can be slightly simplified.

```suggestion
    if let value = dictionary["documentSelector"] {
      self.documentSelector = DocumentSelector(fromLSPArray: value)
    } else {
      self.documentSelector = nil
    }
```